doc/tutorial/web/rest/asab.webservice.md

```diff
@@ -0,0 +1,271 @@
+detail in their respective sections below, we will go from top to bottom:   
+`myrestapi.py`, `./myrestapi/app.py`, `./myrestapi/crud/handler.py`, and `./myrestapi/crud/service.py`  
+While we start working on the microservice, we should have a testing MongoDB instance running, we will touch briefly 
+on how to quickly do that.   
+We will be using Postman to test the API and to generate json collection of available endpoints.
+
+#### myrestapi.py
+
+This is the part, from where we run our app. We begin with the shebang line, that will tell the 
+executing operating system, that we want python to execute this program.  
+Linux: `#!/usr/bin/python3`  
+Windows `#! python3`   
+OS X `#!/usr/bin/env python3`
+
+We follow this up with imports, and all we will need here is our application:   
+`from myrestapi.app import CRUDApp`   
+Next, we instantiate an object of our CRUDApp class in the __main__ of the application, and we run the app:  
+    
+    if __name__ == __main__
+        app = CRUDApp()   
+        app.run()
+        
+#### ./myrestapi/app.py
+
+Here, define our CRUDApp object, we will first need some imports:   
+ 
+    import asab
+    import asab.web
+    import asab.web.rest
+  
+Now we need to add some default configs:    
+ 
+    asab.Config.add_defaults(
+	{
+		'asab:storage': {
+			'type': 'mongodb',
+			'mongodb_uri': 'mongodb://mongouser:mongopassword@mongoipaddress:27017',
+			'mongodb_database': 'mongodatabase'
+		},
+	})
+    
+Next, we describe the class, it inherits from the basic ASAB Application class, but we need to expand 
+it a little (see comments in the code for more details):
+    
+    
+    class CRUDApp(asab.Application):
+
+	    def __init__(self):
+		    super().__init__()
+		    # Circumvent the need for -w flag when starting the app
+		    import asab.storage
+		    self.add_module(asab.web.Module)
+		    self.add_module(asab.storage.Module)
+
+		    # Locate the web service
+		    self.WebService = self.get_service("asab.WebService")
+		    self.WebContainer = asab.web.WebContainer(self.WebService, "web")
+		    self.WebContainer.WebApp.middlewares.append(asab.web.rest.JsonExceptionMiddleware)
+
+		    # Initialize services, we can initialize one, or several API handlers/services here
+		    from .crud.handler import CRUDWebHandler
+		    from .crud.service import CRUDService
+		    self.CRUDService = CRUDService(self)
+		    # We need to pass the CRUDService as an argument, when instantiating the class
+		    self.CRUDWebHandler = CRUDWebHandler(self, self.CRUDService)
+
+This could also be a BSPUMP.Application, enabling us to include a bspump pipeline in our microservice. 
+To do this, inherit instead from bspump.BSPumpApplication and add service, connections and pipelines as 
+usual.
+   
+#### ./myrestapi/crud/handler.py
+
+The handler is where we define the operations our microservice will be able to perform. As the 
```

Comment:
The handler is where HTTP Rest calls are handled and transformed into the actual (internal) service calls. From another perspective, the handler should contain only translation between REST calls and service interface. No actual 'business logic' should be here.